src/process.rs

```diff
@@ -134,11 +137,147 @@
 ///            Expression::Paren(Box::new(Expression::Paren(Box::new(Expression::Letter('z'))))));
 /// # }
 /// ```
+///
+/// ### Sentence
+///
+/// To showcase the use of multiple matchers, we'll use a setence grammar:
+///
+/// ```no_run
+/// # #[macro_use] extern crate pest;
+/// # use pest::prelude::*;
+/// # fn main() {
+/// # impl_rdp! {
+/// # grammar! {
+/// sentence = { word ~ ([" "] ~ word)* }
+/// word     = { letter* }
+/// letter   = { ['a'..'z'] }
+/// # }
+/// # }
+/// # unreachable!();
+/// # }
+/// ```
+///
+/// A very simple AST that works in this case:
+///
+/// ```no_run
+/// # use std::collections::LinkedList;
+/// #[derive(Debug, PartialEq)]
+/// pub enum Node {
+///     Sentence(LinkedList<Node>), // we're using LinkedList because they're more efficient when
+///     Word(LinkedList<Node>),     // using tail recursion
+///     Letter(char)
+/// }
+/// ```
+///
+/// To build the `Token` processor, let's use a bottom up aproach when writing the matchers. Let's
+/// start by building a `word` matcher. We'll call it `_word` in order not to clash with the name
+/// of the rule that also gets defined as a method on the `Parser`.
+///
+/// ```ignore
+/// _word(&self) -> LinkedList<Node> { // return LinkedList<Node> to buil Word with
+///     (&head: letter, @tail) => { // usual tail recursion; &head is captured,
+///         let mut tail = tail;    // tail is recursive
+///         tail.push_front(Node::Letter(head.chars().next().unwrap()));
+///
+///         tail
+///     },
+///     () => {                // if rule above doesn't match, there are no more letters to
```

Comment:
I've thought about it, but I think having a `match` keyword would be confusing since you're not matching against an object, rather against an arbitrary number of items at the start of the queue. Maybe once [slice patterns](https://doc.rust-lang.org/book/slice-patterns.html) get stable it will be worth changing the syntax, but this remains to be seen.